RecoEgamma/EgammaPhotonProducers/src/PhotonProducer.cc

```diff
@@ -417,8 +433,50 @@ void PhotonProducer::fillPhotonCollection(edm::Event& evt,
     showerShape.hcalDepth1OverEcalBc = hcalDepth1OverEcalBc;
     showerShape.hcalDepth2OverEcalBc = hcalDepth2OverEcalBc;
     showerShape.hcalTowersBehindClusters =  TowersBehindClus;
-    newCandidate.setShowerShapeVariables ( showerShape );
-
+    /// fill extra shower shapes
+    const float spp = (!edm::isFinite(locCov[2]) ? 0. : sqrt(locCov[2]));
```

Comment:
I was just following this: https://github.com/cms-sw/cmssw/blob/master/RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc#L586

@Sam-Harper , @bendavid  can comment?